File: cpp/1072. 按列翻转得到最大值等行数.cpp
class Solution {
public:
    int maxEqualRowsAfterFlips(vector<vector<int>>& matrix) {
        unordered_map<string,int> mp;
        int m=matrix.size(),n=matrix[0].size();
        for(int i=0;i<m;i++){
            string t=string(n, '0');
            for(int j=0;j<n;j++){
                t[j]='0'+(matrix[i][0]^matrix[i][j]);
            }
            mp[t]++;
        }
        int ans=0;
        for(auto &[k,v]:mp){
            ans=max(ans,v);
        }
        return ans;
    }
};

File: 刷题/golang/1072. 按列翻转得到最大值等行数.go
func maxEqualRowsAfterFlips(matrix [][]int) int {
	mp := make(map[string]int)
	m, n := len(matrix), len(matrix[0])
	for i := 0; i < m; i++ {
		t := make([]byte, n)
		for j := 0; j < n; j++ {
			t[j] = byte('0' + matrix[i][0] ^ matrix[i][j])
		}
		mp[string(t)]++
	}
	ans := 0
	for _, v := range mp {
		ans = max(ans, v)
	}
	return ans
}
func max(x, y int) int {
	if x > y {
		return x
	}
	return y
}